packages/p2p/src/components/recommend-user/__test__/recommend-user.spec.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { render, screen, fireEvent, within } from '@testing-library/react';
+import RecommendUser from '../recommend-user';
+
+describe('<RecommendUser />', () => {
+    const recommend_user_props = {
+        is_buy_order_for_user: false,
+        is_user_recommended_previously: null,
+        onClickClearRecommendation: jest.fn(),
+        onClickNotRecommended: jest.fn(),
+        onClickRecommended: jest.fn(),
+    };
+
+    it('should render the component with correct message if it is not a buy order for the user and both buttons', () => {
+        render(<RecommendUser {...recommend_user_props} />);
+
+        expect(screen.getByText('Would you recommend this buyer?')).toBeInTheDocument();
+        expect(screen.getAllByRole('button').length).toBe(2);
+    });
+
+    it('should render the component with correct message if it is a buy order for the user and both buttons', () => {
+        render(<RecommendUser {...recommend_user_props} is_buy_order_for_user />);
+
+        expect(screen.getByText('Would you recommend this seller?')).toBeInTheDocument();
+        expect(screen.getAllByRole('button').length).toBe(2);
+    });
+
+    it('should auto select the Yes button if the user was previously recommended', () => {
+        render(<RecommendUser {...recommend_user_props} is_user_recommended_previously={1} />);
+
+        const yesButton = screen.getByRole('button', { name: 'Yes' });
+        const noButton = screen.getByRole('button', { name: 'No' });
+
+        const yesText = within(yesButton).getByText('Yes');
+        const noText = within(noButton).getByText('No');
+
+        expect(yesText).toHaveStyle('color: var(--text-prominent)');
+        expect(noText).toHaveStyle('color: var(--text-less-prominent)');
+    });
+
+    it('should disable the Yes Button after being clicked', () => {
+        render(<RecommendUser {...recommend_user_props} is_user_recommended_previously={1} />);
+
+        const yesButton = screen.getByRole('button', { name: 'Yes' });
+        const yesText = within(yesButton).getByText('Yes');
+
+        fireEvent.click(yesButton);
```

Comment:
nit: Hmm, dont we prefer userEvent than fireEvent? 😆  based on the guidelines. No need to change it 🙏

Reply:
trueee, lets try and bring this into discussion 👀

Reply:
![image](https://github.com/farrah-deriv/deriv-app/assets/125247833/e9f0e2cc-ec18-4ec0-a694-04124802e218)

The different is with `userEvent.click` we combine multiple actions like mouseover, mousedown, and click itself. It used fireEvent under the hood haha. It will be useful for component that need to hover to show the content in desktop and need to click in mobile view like `RecommendedBy`